tests/functional/conftest.py

```diff
@@ -13,3 +16,31 @@ def odf_cli_setup():
         pytest.fail(str(ex))
 
     return odf_cli_runner
+
+
+@pytest.fixture()
+def init_sanity(request, nodes):
+    """
+    Initial Cluster sanity
+    """
+    sanity_helpers = Sanity()
+
+    def finalizer():
+        """
+        Make sure all the nodes are Running and
+        the ceph health is OK at the end of the test
+        """
+
+        # Check if all the nodes are Running otherwise start them
+        log.info("Checking if all the nodes are READY otherwise start them")
+        nodes.restart_nodes_by_stop_and_start_teardown()
+
+        # Check cluster health
+        try:
+            log.info("Making sure ceph health is OK")
+            sanity_helpers.health_check(tries=50, cluster_check=False)
+        except CephHealthException as e:
+            log.error(f"[Error] {e.args}")
+            raise
```

Comment:
do we really need this? IIUC,  this health check is done by default at the end of every test\

Reply:
oh okay! Let me verify that